### PR TITLE
Handling Sharing text & files

### PR DESCRIPTION
**Pull request by MarcoPolo**
This might be the first time you can share a file in chat on mobile?

This does the simplest thing. When you share a file/text and you have a selected conversation, prepare the file/text to be sent to that chat.

This enables attaching images from the keyboard as well.

This removes the broken/buggy kbfs attachment stuff. We could do something similar to this with kbfs. If you are currently looking at a kbfs folder and then you go to some other app and share to keybase, Keybase **prepares** to put the file in the selected folder. Note **prepares**, the user should still be able to cancel – as in the chat flow.

It doesn't give users a chat picker or folder picker for where they want to put the file. If you aren't in a conversation it doesn't do anything. Maybe it should show a toast explaining that you should pick a chat first then share.

I haven't looked at how iOS handles this, but presumably having it fire these new actions will make it just work.

---

Screenshot showing files and sticker from keyboard:

7bb82b7e-8895-434f-82c0-7d69518d6ff3

### CODE REVIEW

File: shared/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java [hunk 1]
```diff
@@ -131,7 +131,10 @@ private String colorSchemeForCurrentConfiguration() {
   @TargetApi(Build.VERSION_CODES.KITKAT)
   protected void onCreate(Bundle savedInstanceState) {
     ReactInstanceManager instanceManager = this.getReactInstanceManager();
-    instanceManager.createReactContextInBackground();
+    if (!instanceManager.hasStartedCreatingInitialContext()) {
```
Comment (MarcoPolo) [inline shared/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: I thought `createReactContextInBg` was a no-op if called twice, but apparently not? RN got mad at me, so I added this check

**Comment by MarcoPolo** [comment]
cc @keybase/react-hackers

**Comment by jzila** [comment]
Hmm, a picker would definitely be better, UX-wise. Although I like the keyboard integration.

**Comment by MarcoPolo** [comment]
> Hmm, a picker would definitely be better, UX-wise. Although I like the keyboard integration.

Agreed. We have something that is almost a picker but it is currently broken: https://github.com/keybase/client/tree/629ab15ab03f9fe44999f7e8be2ce5ca46e72788/shared/fs/send-to-chat/attachment.

This PR makes those changes easier in the future.

The keyboard integration is kind of a hack. In native android you should do something like this https://developer.android.com/guide/topics/text/image-keyboard. If your text input doesn't accept images, then the keyboard sends an intent instead. This PR handles the intent. But that means we don't have a great way of distinguishing whether a send Intent came from the keyboard or somewhere else.

**Pull request closed without merging**